crates/proto/src/op/header.rs

```diff
@@ -514,7 +514,7 @@ impl BinEncodable for Header {
         q_opcd_a_t_r |= u8::from(self.op_code) << 3;
         q_opcd_a_t_r |= if self.authoritative { 0x4 } else { 0x0 };
         q_opcd_a_t_r |= if self.truncation { 0x2 } else { 0x0 };
-        q_opcd_a_t_r |= if self.recursion_desired { 0x1 } else { 0x0 };
+        q_opcd_a_t_r |= u8::from(self.recursion_desired); // 0x1 or 0x0
```

Comment:
Personally I always like to avoid `if` for boolean to direct numeric value, but leaved the comment because explicit is better.

Reply:
See my comment below, if this is a clippy warning, I think we should just disable the check...

Reply:
https://rust-lang.github.io/rust-clippy/master/#bool_to_int_with_if I am making a commit  to disable it.

Reply:
done! ;)